controlpanel/api/rules.py

```diff
@@ -49,8 +61,8 @@ def is_app_admin(user, obj):
 add_perm('api.destroy_app', is_authenticated & is_superuser)
 add_perm('api.add_app_customer', is_authenticated & is_app_admin)
 add_perm('api.remove_app_customer', is_authenticated & is_app_admin)
-add_perm('api.add_app_admin', is_authenticated & is_superuser)
-add_perm('api.revoke_app_admin', is_authenticated & is_superuser)
+add_perm('api.add_app_admin', is_authenticated & is_app_admin)
+add_perm('api.revoke_app_admin', is_authenticated & is_app_admin)
```

Comment:
Random thought (not related to your changes): These predicates seems to repeat `is_authenticated ` a lot, maybe it's to be safe in case the other predicated forget to check the user is authenticated before carrying one with other checks?

From an high level perspective of course is a user is a `superuser` for example, I'd imply that it's also authenticated but logic and the code may disagree 😬

Reply:
IIRC `is_authenticated` is built into the module we're using and we're creating new predicates by explicitly combining two existing ones (one built into the module, the other defined by us).